src/p2p/behaviour.rs

```diff
@@ -104,7 +106,14 @@ impl<Types: IpfsTypes> NetworkBehaviourEventProcess<KademliaEvent> for Behaviour
                         );
                     }
                     Bootstrap(Err(BootstrapError::Timeout { .. })) => {
-                        warn!("kad: failed to bootstrap");
+                        warn!("kad: timed out while trying to bootstrap");
+
+                        if self.kademlia.query(&id).is_none() {
```

Comment:
Just to clarify, this:

```
if self.kademlia.query(&id).is_none() {
```

Is done to to guard against the query returning a result but still being ongoing or something?

Reply:
yes, exactly that - some queries return partial results, and we don't want to finish the subscription until it's complete